lib/binary_to_decimal.rb

```diff
@@ -5,5 +5,36 @@
 # Calculate  and return the decimal value for this binary number using
 # the algorithm you devised in class.
 def binary_to_decimal(binary_array)
-  raise NotImplementedError
+  index_position = binary_array.length - 1
+  decimal_total = 0
+  
+  binary_array.length.times do |i|
+    decimal_total += binary_array[index_position] * (2**i)
+    
+    index_position -= 1
+  end
+  
+  return decimal_total
+  
+end
+
+def decimal_to_binary(decimal_number)
+  binary_array = []
+  binary_value = 128
+  new_value = decimal_number
+  
+  8.times do |i|
+    if new_value >= binary_value
+      binary_array << 1
+      new_value -= binary_value
+      binary_value /= 2
```

Comment:
Minor issue:  since you are doing `binary_value /= 2` in both the if and else, you could move this to after the if-else and just have 1 line.